website/docs/r/kubernetes_cluster_node_pool.html.markdown

```diff
@@ -312,6 +312,12 @@ A `upgrade_settings` block supports the following:
 
 * `max_surge` - (Required) The maximum number or percentage of nodes which will be added to the Node Pool size during an upgrade.
```

Comment:
This shall be `Optional`? Also, this changes the semantic a bit as I assume it is still required when `max_unavailable` is not specified?